lib/open_food_network/proxy_order_syncer.rb

```diff
@@ -56,9 +56,10 @@ def remove_orphaned_proxy_orders!
     end
 
     def orphaned_proxy_orders
-      order_cycle_ids = active_or_complete_or_in_range_order_cycle_ids
-      return proxy_orders unless order_cycle_ids.any?
-      proxy_orders.where('order_cycle_id NOT IN (?)', order_cycle_ids)
+      orphaned = proxy_orders.where(placed_at: nil)
+      order_cycle_ids = in_range_order_cycles.pluck(:id)
+      return orphaned unless order_cycle_ids.any?
```

Comment:
I was just wondering if we could skip this line, but SQL seems to be stupid enough to not accept an empty collection: `order_cycle_id NOT IN ()` -> `PG::Error: ERROR:  syntax error at or near ")"`. And ActiveRecord deals with it by generating `order_cycle_id NOT IN (NULL)` which doesn't return anything. Any operation with `NULL` always returns `NULL`.

One way to make SQL deal with an empty set is the empty result from a sub query:

```ruby
order_cycle_ids = in_range_order_cycles.select(:id)
orphaned.where("order_cycle_id NOT IN (#{order_cycle_ids.to_sql})")
```

This is saving a query from time to time, but I'm not sure if it's easier to read. I think I'm micro-optimising here. But it would make sense if we used Rails 4 already (saving even another line):

```ruby
orphaned.where.not(order_cycle_id: in_range_order_cycles)
```
Can't wait to upgrade.

Reply:
Well, I think you will have to @mkllnk :trollface: